ocaml/xapi/xapi_pusb.ml

```diff
@@ -88,3 +88,54 @@ let scan_thread ~__context =
   in
   start_thread f;
   scan ~__context
+
+let get_sm_usb_path ~__context vdi =
+  try
+    let vdi_sc = Db.VDI.get_sm_config ~__context ~self:vdi in
+    List.assoc Xapi_globs.usb_path vdi_sc
+  with _ -> ""
+
+let set_passthrough_enabled ~__context ~self ~value =
+  Mutex.execute mutex (fun () ->
+    match value with
+     | true ->
+       (* Remove the vdi records which 'usb_path' in sm-config has the
+          same value with the field 'path' in PUSB. *)
+       Db.VDI.get_all_records ~__context |>
```

Comment:
we should restrict this get_all_records to just the udevSR. You can do this by doing `get_all_records_where` with an expression that matches the SR reference of the udevSR. This SR reference can be obtained by looking for an SR record that has the correct `type` field.

The issue is that there may be thousands of VDI records, so the more we can do to filter them the better.

Reply:
updated.